src/mds/Server.cc

```diff
@@ -1122,8 +1128,11 @@ void Server::recover_filelocks(CInode *in, bufferlist locks, int64_t client)
  * to trim some caps, and consequently unpin some inodes in the MDCache so
  * that it can trim too.
  */
-void Server::recall_client_state(void)
+void Server::recall_client_state(double ratio, bool report_acks)
 {
+  MClientSession *m;
+  static MDSGatherBuilder gather(g_ceph_context);
```

Comment:
why static?

Reply:
I saw MDS crash without it -

```
(gdb)
1155      set<Session*> sessions;
(gdb)
1133      MDSGatherBuilder gather(g_ceph_context);
(gdb)

Program received signal SIGABRT, Aborted.
0x00007f7fd283b277 in __GI_raise (sig=sig@entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56        return INLINE_SYSCALL (tgkill, 3, pid, selftid, sig);
(gdb)
handle_fatal_signal (signum=6)
    at /home/centos/repos/ceph/src/global/signal_handler.cc:92
92      {
(gdb) c
Continuing.
```
I do not have a higher understanding of how MDS runs as a daemon but I saw the line being executed multiple times. So, the making it `static` seemed like a natural solution.